### PR TITLE
Removes old proof code

### PR DESCRIPTION
**Pull request by HanielB**
This deletes much of the old proof code. Basically everything but the minimal necessary infra-structure for producing unsat cores. That includes dependency tracking in preprocessing, the prop engine proof and the unsat core computation code in the old proof manager.

It also cleans interfaces that were using old-proof-code specific constructs (such as `LemmaProofRecipe`). When possible or when it made sense standalone local proof production code was kept, active or not (such is in the equality engine and in the arithmetic solver).

### CODE REVIEW

File: src/preprocessing/passes/quantifier_macros.cpp [hunk 1]
```diff
@@ -432,9 +439,10 @@ Node QuantifierMacros::simplify( Node n ){
           for( unsigned i=0; i<children.size(); i++ ){
             Node etc = TypeNode::getEnsureTypeCondition( children[i], tno[i] );
             if( etc.isNull() ){
-              //if this does fail, we are incomplete, since we are eliminating quantified formula corresponding to op, 
+              // if this does fail, we are incomplete, since we are eliminating
+              // quantified formula corresponding to op,
               //  and not ensuring it applies to n when its types are correct.
-              //Assert( false );
+              // Assert( false );
```
Comment (ajreynol) [inline src/preprocessing/passes/quantifier_macros.cpp:445]: Delete?

File: src/proof/clause_id.h [hunk 2]
```diff
@@ -28,6 +28,12 @@ namespace CVC4 {
  */
 typedef unsigned ClauseId;
 
+const ClauseId ClauseIdEmpty(-1);
```
Comment (ajreynol) [inline src/proof/clause_id.h:31]: What are these? Can you add a some quick (basic) documentation here? E.g. "Temporary, to do delete when X" or whatever.

File: test/unit/prop/cnf_stream_white.h [hunk 3]
```diff
@@ -240,18 +241,17 @@ class CnfStreamWhite : public CxxTest::TestSuite {
   //              kind::ITE,
   //              d_nodeManager->mkVar(d_nodeManager->booleanType()),
   //              d_nodeManager->mkVar(d_nodeManager->integerType()),
-  //              d_nodeManager->mkVar(d_nodeManager->integerType())
-  //          ),
-  //          d_nodeManager->mkVar(d_nodeManager->integerType())
-  //                            ), false, false, RULE_INVALID, Node::null());
-  //
-  //}
+  //              d_nodeManager->mkVar(d_nodeManager->integerType())),
```
Comment (ajreynol) [inline test/unit/prop/cnf_stream_white.h:244]: Delete this block

File: test/regress/regress1/quantifiers/dump-inst-proof.smt2 [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ; REQUIRES: proof
-; COMMAND-LINE: --dump-instantiations --proof --print-inst-full
+; COMMAND-LINE: --dump-instantiations --produce-unsat-cores --print-inst-full
```
Comment (ajreynol) [inline test/regress/regress1/quantifiers/dump-inst-proof.smt2:2]: I like this change.  It didn't make much sense to use `--proof` for this feature.

File: src/theory/uf/equality_engine_types.h [hunk 5]
```diff
@@ -63,20 +63,24 @@ static const EqualityEdgeId null_edge = (EqualityEdgeId)(-1);
...
-  /** Equality was merged to false, due to both sides of equality being a constant */
+  /** Terms were merged due to theory reasoning */
   MERGED_THROUGH_CONSTANTS,
-  /** (for proofs only) Equality was merged due to transitivity */
+  /** Terms were merged due to transitivity */
   MERGED_THROUGH_TRANS,
-
-  /** Reason types beyond this constant are theory specific reasons */
-  NUMBER_OF_MERGE_REASONS
+  /** Terms were merged due to arrays read-over-write */
```
Comment (ajreynol) [inline src/theory/uf/equality_engine_types.h:78]: These shouldn't have to be added here.  Can you try to delete these as well?  Or perhaps at least mark "//!!!!!!!! temporary".  I believe I can delete them if you need.

File: src/theory/theory.cpp [hunk 6]
```diff
@@ -23,6 +23,7 @@
 
 #include "base/check.h"
 #include "expr/node_algorithm.h"
+#include "options/smt_options.h"
```
Comment (ajreynol) [inline src/theory/theory.cpp:26]: Why is this needed?

File: src/theory/arith/theory_arith_private.cpp [hunk 7]
```diff
@@ -4716,7 +4719,8 @@ bool TheoryArithPrivate::rowImplicationCanBeApplied(RowIndex ridx, bool rowUp, C
       }
       Node implication = implied->externalImplication(explain);
       Node clause = flattenImplication(implication);
-      PROOF(if (d_containing.d_proofRecorder
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/theory_arith_private.cpp:4722]: Delete?

File: src/theory/arith/theory_arith_private.cpp [hunk 8]
```diff
@@ -4700,8 +4700,11 @@ bool TheoryArithPrivate::rowImplicationCanBeApplied(RowIndex ridx, bool rowUp, C
   if( !assertedToTheTheory && canBePropagated && !hasProof ){
     ConstraintCPVec explain;
 
-    PROOF(d_farkasBuffer.clear());
-    RationalVectorP coeffs = NULLPROOF(&d_farkasBuffer);
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/theory_arith_private.cpp:4703]: Delete?

File: src/theory/arith/theory_arith_private.cpp [hunk 9]
```diff
@@ -1923,7 +1924,8 @@ void TheoryArithPrivate::outputConflicts(){
       ++conflicts;
       Debug("arith::conflict") << "d_conflicts[" << i << "] " << conflict
                                << " has proof: " << hasProof << endl;
-      PROOF(if (d_containing.d_proofRecorder && confConstraint->hasFarkasProof()
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/theory_arith_private.cpp:1927]: Delete?  @alex-ozdemir  may know best about this.  Wondering if this is required on proof-new.

File: src/theory/arith/constraint.h [hunk 10]
```diff
@@ -877,9 +872,12 @@ class Constraint {
   }
 
   inline RationalVectorCP getFarkasCoefficients() const {
-    return NULLPROOF(getConstraintRule().d_farkasCoefficients);
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/constraint.h:875]: Ditto.

File: src/theory/arith/constraint.cpp [hunk 11]
```diff
@@ -1374,12 +1418,11 @@ void Constraint::impliedByFarkas(const ConstraintCPVec& a, RationalVectorCP coef
   AntecedentId antecedentEnd = d_database->d_antecedents.size() - 1;
 
   RationalVectorCP coeffsCopy;
-  if(PROOF_ON()){
-    Assert(coeffs != RationalVectorCPSentinel);
-    coeffsCopy = new RationalVector(*coeffs);
-  } else {
-    coeffsCopy = RationalVectorCPSentinel;
-  }
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/constraint.cpp:1421]: Ditto.

File: src/theory/arith/constraint.cpp [hunk 12]
```diff
@@ -1359,10 +1402,11 @@ void Constraint::impliedByFarkas(const ConstraintCPVec& a, RationalVectorCP coef
   Assert(negationHasProof() == nowInConflict);
   Assert(allHaveProof(a));
 
-  Assert(PROOF_ON() == (coeffs != RationalVectorCPSentinel));
-  // !PROOF_ON() => coeffs == RationalVectorCPSentinel
-  //  PROOF_ON() => coeffs->size() == a.size() + 1
-  Assert(!PROOF_ON() || coeffs->size() == a.size() + 1);
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/constraint.cpp:1405]: Ditto.

File: src/theory/arith/constraint.cpp [hunk 13]
```diff
@@ -1213,7 +1257,7 @@ void Constraint::impliedByUnate(ConstraintCP imp, bool nowInConflict){
   AntecedentId antecedentEnd = d_database->d_antecedents.size() - 1;
 
   RationalVectorP coeffs;
-  if(PROOF_ON()){
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/constraint.cpp:1260]: Ditto.

File: src/theory/arith/constraint.cpp [hunk 14]
```diff
@@ -701,109 +702,153 @@ bool Constraint::wellFormedFarkasProof() const {
   if(antecedent  == NullConstraint) { return false; }
 
 #if IS_PROOFS_BUILD
-  if(!PROOF_ON()){ return cr.d_farkasCoefficients == RationalVectorCPSentinel; }
-  Assert(PROOF_ON());
+  return cr.d_farkasCoefficients == RationalVectorCPSentinel;
+#if 0
```
Comment (ajreynol) [inline src/theory/arith/constraint.cpp:706]: Ditto.

File: src/proof/clause_id.h (hunk 2 above)
Comment (HanielB) [reply src/proof/clause_id.h:31]: These are used to control registration of clauses in the old sat proof (which is used for unsat cores, so we have to keep it for now). I'll add a comment. As a rule of thumb, everything that is in `/proof` currently will be deleted in its current form.

File: src/theory/theory.cpp (hunk 6 above)
Comment (HanielB) [reply src/theory/theory.cpp:26]: Because there is a call to `options::produceModels()` in the file. This was not necessary before because the inclusion-chain to `theory.h` had `smt_options.h` before (not sure where) but does not any longer.

**Comment by HanielB** [comment]
Answering the general concerns:

- (1) : There was a lot of code in the arithmetic solver to make it proof producing that I was afraid of deleting. The one who would know better how it relates to the plan for `proof-new` would be @alex-ozdemir, so to be on the safe side I kept it while not executing. But we can go ahead and delete.

- (2) : I think this is not as problematic. The original formatting is strange and any modification leads to clang-format changing things significantly. I think it makes sense to just apply them. I can handle the merge as they will be with the code I've modified in proof-new.

File: src/theory/uf/equality_engine_types.h (hunk 5 above)
Comment (HanielB) [reply src/theory/uf/equality_engine_types.h:78]: The array solver sends these reasons when it's asserting equalities. Previously it would dynamically generate these three, so there was some dedicated code for allowing for that. By making them static the handling of reasons is simpler.

Comment (ajreynol) [reply src/theory/uf/equality_engine_types.h:78]: just write temporary then, I'll remove on proof-new.

Comment (HanielB) [reply src/theory/uf/equality_engine_types.h:80]: OK, done.

File: src/theory/theory.cpp (hunk 6 above)
Comment (ajreynol) [reply src/theory/theory.cpp:26]: Makes sense.

**Comment by ajreynol** [comment]
Regarding the merge: I think we may need to collaborate.  I'm estimating there will be several non-trivial conflicts in code I've modified on proof-new.

**Comment by alex-ozdemir** [comment]
An initial comment: the `ArithProofRecorder` class (and its use-sites) should be removed. The old machinery used them, but the new machinery does not (since the new machinery builds proofs at lemma/conflict time, it doesn't need a separate persistent storage for this information: this information ends up in an `EagerProofGenerator`.

**Comment by HanielB** [comment]
Good to know, @alex-ozdemir! I've update the PR to remove the ArithProofRecorder and moreover the dead code I had kept.

File: src/smt/term_formula_removal.cpp [hunk 15]
```diff
@@ -402,13 +407,14 @@ Node RemoveTermFormulas::replace(TNode node, bool inQuant, bool inTerm) const {
...
 
-// returns true if the children of node should be considered nested terms 
+// returns true if the children of node should be considered nested terms
 bool RemoveTermFormulas::hasNestedTermChildren( TNode node ) {
-  return theory::kindToTheoryId(node.getKind())!=theory::THEORY_BOOL && 
-         node.getKind()!=kind::EQUAL && node.getKind()!=kind::SEP_STAR && 
-         node.getKind()!=kind::SEP_WAND && node.getKind()!=kind::SEP_LABEL && 
-         node.getKind()!=kind::BITVECTOR_EAGER_ATOM;
-         // dont' worry about FORALL or EXISTS (handled separately)
+  return theory::kindToTheoryId(node.getKind()) != theory::THEORY_BOOL
```
Comment (ajreynol) [inline src/smt/term_formula_removal.cpp:412]: Note that this block of code has been moved on proof-new.  Thus you will get a merge conflict unless you revert this formatting.

**Comment by HanielB** [comment]
I would like to have the approval of @alex-ozdemir as well. All the deletions were somewhat straightforward except for the arithmetic part.

File: src/theory/arith/callbacks.cpp [hunk 16]
```diff
@@ -87,37 +87,25 @@ void FarkasConflictBuilder::reset(){
   d_consequent = NullConstraint;
   d_constraints.clear();
   d_consequentSet = false;
-  PROOF(d_farkas.clear());
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:90]: Ah, I see now that I should have been more clear earlier about how arith records farkas coefficients, and what we need to keep/remove.

Prior to this PR, arith maintained farkas coefficients in two locations:
a) In data structures in the `src/theory/arith` directory (e.g. `FarkasConflictBuilder` and `ConstraintRule`).
b) In data structures in the `src/proof` directory (e.g. `ArithProofRecorder`).

The former may be context-dependent. The latter were persistent.

We can remove the latter. They're replace by eager proof generators, etc., in the new proof system. **We cannot remove the former**: the new system uses them.

Generally, in the core of arithmetic, this means that we should treat proofs as "always on" and not trim code that is guarded by compile-time proof flags. Thus, this command should be kept, just `d_farkas.clear()`.

On the other hand, we can remove all of `src/proof/arith_*`, and direct references to such machinery in `src/theory/arith/`.

File: src/theory/arith/callbacks.cpp [hunk 17]
```diff
@@ -87,37 +87,25 @@ void FarkasConflictBuilder::reset(){
...
 }
 
 /* Adds a constraint to the constraint under construction. */
 void FarkasConflictBuilder::addConstraint(ConstraintCP c, const Rational& fc){
-  Assert(
-      !PROOF_ON()
-      || (!underConstruction() && d_constraints.empty() && d_farkas.empty())
-      || (underConstruction() && d_constraints.size() + 1 == d_farkas.size()));
-  Assert(PROOF_ON() || d_farkas.empty());
+  Assert(d_farkas.empty());
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:95]:
```suggestion
  Assert((!underConstruction() && d_constraints.empty() && d_farkas.empty())
      || (underConstruction() && d_constraints.size() + 1 == d_farkas.size()));
```

Same as above, we should treat proofs as "always on", but otherwise not modify the core of how arith computes and stores farkas coefficients.

File: src/theory/arith/callbacks.cpp [hunk 18]
```diff
@@ -87,37 +87,25 @@ void FarkasConflictBuilder::reset(){
...
 
   if(d_consequent == NullConstraint){
     d_consequent = c;
   } else {
     d_constraints.push_back(c);
   }
-  PROOF(d_farkas.push_back(fc););
-  Assert(!PROOF_ON() || d_constraints.size() + 1 == d_farkas.size());
-  Assert(PROOF_ON() || d_farkas.empty());
+  Assert(d_farkas.empty());
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:103]:
```suggestion
  d_farkas.push_back(fc);
  Assert(d_constraints.size() + 1 == d_farkas.size());
```
ditto

File: src/theory/arith/callbacks.cpp [hunk 19]
```diff
@@ -87,37 +87,25 @@ void FarkasConflictBuilder::reset(){
...
 
 void FarkasConflictBuilder::addConstraint(ConstraintCP c, const Rational& fc, const Rational& mult){
   Assert(!mult.isZero());
-  if(PROOF_ON() && !mult.isOne()){
-    Rational prod = fc * mult;
-    addConstraint(c, prod);
-  }else{
-    addConstraint(c, fc);
-  }
+  addConstraint(c, fc);
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:108]:
```suggestion
  if(!mult.isOne()){
    Rational prod = fc * mult;
    addConstraint(c, prod);
  }else{
    addConstraint(c, fc);
  }
```
ditto

File: src/theory/arith/callbacks.cpp [hunk 20]
```diff
@@ -132,7 +120,6 @@ void FarkasConflictBuilder::makeLastConsequent(){
     ConstraintCP last = d_constraints.back();
     d_constraints.back() = d_consequent;
     d_consequent = last;
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:122]:
```suggestion
    d_consequent = last;
    std::swap( d_farkas.front(), d_farkas.back() );
```

File: src/theory/arith/callbacks.cpp [hunk 21]
```diff
@@ -144,15 +131,11 @@ void FarkasConflictBuilder::makeLastConsequent(){
 ConstraintCP FarkasConflictBuilder::commitConflict(){
   Assert(underConstruction());
   Assert(!d_constraints.empty());
-  Assert(
-      !PROOF_ON()
-      || (!underConstruction() && d_constraints.empty() && d_farkas.empty())
-      || (underConstruction() && d_constraints.size() + 1 == d_farkas.size()));
-  Assert(PROOF_ON() || d_farkas.empty());
+  Assert(d_farkas.empty());
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:134]:
```suggestion
  Assert((!underConstruction() && d_constraints.empty() && d_farkas.empty())
      || (underConstruction() && d_constraints.size() + 1 == d_farkas.size()));
```
ditto

File: src/theory/arith/callbacks.cpp [hunk 22]
```diff
@@ -144,15 +131,11 @@ void FarkasConflictBuilder::makeLastConsequent(){
...
   Assert(d_consequentSet);
 
   ConstraintP not_c = d_consequent->getNegation();
-  RationalVectorCP coeffs = NULLPROOF(&d_farkas);
+  RationalVectorCP coeffs = nullptr;
```
Comment (alex-ozdemir) [inline src/theory/arith/callbacks.cpp:138]:
```suggestion
  RationalVectorCP coeffs = &d_farkas;
```

File: src/theory/arith/constraint.cpp [hunk 23]
```diff
@@ -648,8 +647,7 @@ ConstraintCP ConstraintDatabase::getAntecedent (AntecedentId p) const {
 
 
 void ConstraintRule::print(std::ostream& out) const {
-  
-  RationalVectorCP coeffs = NULLPROOF(d_farkasCoefficients);
+  RationalVectorCP coeffs = nullptr;
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.cpp:650]:
```suggestion
  RationalVectorCP coeffs = d_farkasCoefficients;
```

File: src/theory/arith/constraint.cpp [hunk 24]
```diff
@@ -701,109 +699,9 @@ bool Constraint::wellFormedFarkasProof() const {
...
-  Debug("constraints::wffp") << "final sum: " << lhs << " <= " << rhs << endl;
-  // 0 = lhs <= rhs < 0
-  return (lhs.isNull() || (Constant::isMember(lhs) && Constant(lhs).isZero()))
-         && rhs.sgn() < 0;
-
-#else  /* IS_PROOFS_BUILD */
-  return true;
+  return cr.d_farkasCoefficients == RationalVectorCPSentinel;
 #endif /* IS_PROOFS_BUILD */
+  return true;
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.cpp:704]: Ditto: assume that `IS_PROOFS_BUILD` is true here.

File: src/theory/arith/constraint.cpp [hunk 25]
```diff
@@ -1213,18 +1110,7 @@ void Constraint::impliedByUnate(ConstraintCP imp, bool nowInConflict){
...
-    Rational first(sgns.first);
-    Rational second(sgns.second);
-
-    coeffs = new RationalVector();
-    coeffs->push_back(first);
-    coeffs->push_back(second);
-  } else {
-    coeffs = RationalVectorPSentinel;
-  }
+  coeffs = RationalVectorPSentinel;
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.cpp:1113]:
```suggestion
  std::pair<int, int> sgns = unateFarkasSigns(getNegation(), imp);

  Rational first(sgns.first);
  Rational second(sgns.second);

  coeffs = new RationalVector();
  coeffs->push_back(first);
  coeffs->push_back(second);
```

File: src/theory/arith/constraint.cpp [hunk 26]
```diff
@@ -1359,10 +1245,7 @@ void Constraint::impliedByFarkas(const ConstraintCPVec& a, RationalVectorCP coef
   Assert(negationHasProof() == nowInConflict);
   Assert(allHaveProof(a));
 
-  Assert(PROOF_ON() == (coeffs != RationalVectorCPSentinel));
-  // !PROOF_ON() => coeffs == RationalVectorCPSentinel
-  //  PROOF_ON() => coeffs->size() == a.size() + 1
-  Assert(!PROOF_ON() || coeffs->size() == a.size() + 1);
+  Assert(coeffs == RationalVectorCPSentinel);
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.cpp:1248]:
```suggestion
  Assert(coeffs != RationalVectorCPSentinel);
  Assert(coeffs->size() == a.size() + 1);
```

File: src/theory/arith/constraint.h [hunk 27]
```diff
@@ -339,7 +338,7 @@ struct ConstraintRule {
     , d_proofType(pt)
     , d_antecedentEnd(antecedentEnd)
   {
-    Assert(PROOF_ON() || coeffs == RationalVectorCPSentinel);
+    Assert(coeffs == RationalVectorCPSentinel);
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.h:341]:
```suggestion
```

File: src/theory/arith/constraint.h [hunk 28]
```diff
@@ -789,10 +788,6 @@ class Constraint {
       ConstraintP constraint = crp->d_constraint;
       Assert(constraint->d_crid != ConstraintRuleIdSentinel);
       constraint->d_crid = ConstraintRuleIdSentinel;
-
-      PROOF(if (crp->d_farkasCoefficients != RationalVectorCPSentinel) {
-        delete crp->d_farkasCoefficients;
-      });
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.h:795]: Ditto: we should keep this branch.

File: src/theory/arith/constraint.h [hunk 29]
```diff
@@ -876,10 +871,8 @@ class Constraint {
     return getConstraintRule().d_antecedentEnd;
   }
 
-  inline RationalVectorCP getFarkasCoefficients() const {
-    return NULLPROOF(getConstraintRule().d_farkasCoefficients);
-  }
-  
+  inline RationalVectorCP getFarkasCoefficients() const { return nullptr; }
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.h:874]:
```suggestion
  inline RationalVectorCP getFarkasCoefficients() const { return getConstraintRule().d_farkasCoefficients; }
```

File: src/theory/arith/linear_equality.cpp [hunk 30]
```diff
@@ -510,11 +510,11 @@ void LinearEqualityModule::propagateBasicFromRow(ConstraintP c){
   RowIndex ridx = d_tableau.basicToRowIndex(basic);
 
   ConstraintCPVec bounds;
-  RationalVectorP coeffs = NULLPROOF(new RationalVector());
+  RationalVectorP coeffs = nullptr;
```
Comment (alex-ozdemir) [inline src/theory/arith/linear_equality.cpp:513]:
```suggestion
  RationalVectorP coeffs = new RationalVector();
```

File: src/theory/arith/theory_arith.cpp [hunk 31]
```diff
@@ -41,7 +41,6 @@ TheoryArith::TheoryArith(context::Context* c,
       d_internal(
           new TheoryArithPrivate(*this, c, u, out, valuation, logicInfo, pnm)),
       d_ppRewriteTimer("theory::arith::ppRewriteTimer"),
-      d_proofRecorder(nullptr),
```
Comment (alex-ozdemir) [inline src/theory/arith/theory_arith.cpp:44]: Great! This is an example of a reference to the `src/proof` tree, and, thus, it should be removed :+1: .

File: src/theory/arith/theory_arith_private.cpp [hunk 32]
```diff
@@ -1923,35 +1924,6 @@ void TheoryArithPrivate::outputConflicts(){
...
-        }
-
-        Assert(conflict.getNumChildren() == pf.d_farkasCoefficients->size());
-        if (confConstraint->hasSimpleFarkasProof()
-            && confConstraint->getNegation()->isPossiblyTightenedAssumption())
-        {
-          d_containing.d_proofRecorder->saveFarkasCoefficients(
-              conflictInFarkasCoefficientOrder, pf.d_farkasCoefficients);
-        }
-      })
```
Comment (alex-ozdemir) [inline src/theory/arith/theory_arith_private.cpp:1954]: Great! An example of code that's just feeding information into `ArithProofRecorder`. This should be removed :+1: .

File: src/theory/arith/theory_arith_private.cpp [hunk 33]
```diff
@@ -4699,9 +4669,7 @@ bool TheoryArithPrivate::rowImplicationCanBeApplied(RowIndex ridx, bool rowUp, C
 
   if( !assertedToTheTheory && canBePropagated && !hasProof ){
     ConstraintCPVec explain;
-
-    PROOF(d_farkasBuffer.clear());
-    RationalVectorP coeffs = NULLPROOF(&d_farkasBuffer);
+    RationalVectorP coeffs = nullptr;
```
Comment (alex-ozdemir) [inline src/theory/arith/theory_arith_private.cpp:4672]:
```suggestion
    d_farkasBuffer.clear();
    RationalVectorP coeffs = &d_farkasBuffer;
```

File: src/theory/arith/theory_arith_private.cpp [hunk 34]
```diff
@@ -4716,38 +4684,6 @@ bool TheoryArithPrivate::rowImplicationCanBeApplied(RowIndex ridx, bool rowUp, C
...
-        Assert(conflictInFarkasCoefficientOrder.getNumChildren()
-               == coeffs->size());
-        if (std::all_of(explain.begin(), explain.end(), [](ConstraintCP c) {
-              return c->isAssumption() || c->hasIntTightenProof();
-            }))
-        {
-          d_containing.d_proofRecorder->saveFarkasCoefficients(
-              conflictInFarkasCoefficientOrder, coeffs);
-        }
-      })
```
Comment (alex-ozdemir) [inline src/theory/arith/theory_arith_private.cpp:4750]: Great! This should also be removed :+1:

File: src/theory/arith/callbacks.cpp (hunk 16 above)
Comment (HanielB) [reply src/theory/arith/callbacks.cpp:90]: I see, thanks! At first I had put most of this code under `#if 0 ... #endif` statements because it looked important. :) I have applied the changes you suggested, PTAL.

File: src/theory/arith/constraint.cpp (hunk 24 above)
Comment (HanielB) [reply src/theory/arith/constraint.cpp]: In the code block executed when `IS_PROOFS_BUILD` holds, the statement 
```
  return cr.d_farkasCoefficients == RationalVectorCPSentinel;
```
would be executed only if proofs are off, which we should assume is never the case, if I understand correctly. In that case I assume that the code that should survive is the one after, in the old code, `Assert(PROOF_ON());`?

**Comment by HanielB** [comment]
@alex-ozdemir I noticed after the changes that unless everything involving d_farkasCoefficients is guarded with the compile flag we will have issues. So in the interest of simplicity I left its declaration unguarded, which will have the rest of the code as a no-op in a non-proofs-build, if I'm not mistaken. What do you think?

File: src/theory/arith/constraint.cpp [hunk 35]
```diff
@@ -551,46 +550,48 @@ bool Constraint::hasTrichotomyProof() const {
...
+      if (antecdent == NullConstraint)
+      {
+        break;
+      }
+      antecdent->printProofTree(out, depth + 1);
     }
-    antecdent->printProofTree(out, depth + 1);
   }
-#else  /* IS_PROOFS_BUILD */
   out << "Cannot print proof. This is not a proof build." << endl;
```
Comment (alex-ozdemir) [inline src/theory/arith/constraint.cpp:594]:
```suggestion
```

Should this be removed?

Comment (HanielB) [reply src/theory/arith/constraint.cpp:595]: So that we keep things synced with proof-new I've added a return statement in the end of the `if` block and kept the prenting message.
